### ISSUE TITLE
[bug]: 2024.7.1 backend docker image start failed with error "listen tcp :80: bind: permission denied"

### ISSUE THREAD

**Issue by StoneMoe** [hoppscotch/hoppscotch#4264]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current behavior

When I followed the documetation to run self-host hoppscotch with AIO container option, the container log says:
```
App/Admin Dashboard Caddy | Error: loading initial config: loading new config: http app module: start: listening on :80: listen tcp :80: bind: permission denied
```
seems like Caddy somehow decided to bind to port 80 and failed due to permission issue.

then I also tried to run it with the separated service containers, and turns out `admin` and `frontend` container started successfully, but the `backend` container reported the exact same error log.

### Steps to reproduce

1. fresh environment
2. run with AIO container image, with the compose and env file below
3. see logs report port binding error

compose.yml:
```
version: '3.8'

services:
  frontend:
    image: hoppscotch/hoppscotch-frontend:2024.7.1
    ports:
      - "3000:3000"
    env_file:
      - stack.env
    restart: unless-stopped

  backend:
    image: hoppscotch/hoppscotch-backend:2024.7.1
    ports:
      - "3170:3170"
    env_file:
      - stack.env
    restart: unless-stopped

  admin:
    image: hoppscotch/hoppscotch-admin:2024.7.1
    ports:
      - "3100:3100"
    env_file:
      - stack.env
    restart: unless-stopped
  db:
    image: postgres:16.4
    restart: always
    shm_size: 128mb
    environment:
      POSTGRES_PASSWORD: hoppscotch
      POSTGRES_USER: hoppscotch

```

stack.env:
```
DATABASE_URL=postgresql://hoppscotch:hoppscotch@db:5432/hoppscotch # or replace with your database URL
JWT_SECRET=abcdefg0000000
TOKEN_SALT_COMPLEXITY=10
MAGIC_LINK_TOKEN_VALIDITY=3
REFRESH_TOKEN_VALIDITY=604800000 # Default validity is 7 days (604800000 ms) in ms
ACCESS_TOKEN_VALIDITY=86400000 # Default validity is 1 day (86400000 ms) in ms
SESSION_SECRET=hijklmn1111111
ALLOW_SECURE_COOKIES=false
REDIRECT_URL=http://hop.home.lan:3000
WHITELISTED_ORIGINS=http://hop.home.lan:3170,http://hop.home.lan:3000,http://hop.home.lan:3100
VITE_ALLOWED_AUTH_PROVIDERS=EMAIL
MAILER_SMTP_ENABLE=true
MAILER_USE_CUSTOM_CONFIGS=false
MAILER_ADDRESS_FROM=<hoppscotch@home.lan>
MAILER_SMTP_URL=
RATE_LIMIT_TTL=60 # In seconds
RATE_LIMIT_MAX=100 # Max requests per IP
VITE_BASE_URL=http://hop.home.lan:3000
VITE_SHORTCODE_BASE_URL=http://hop.home.lan:3000
VITE_ADMIN_URL=http://hop.home.lan:3100
VITE_BACKEND_GQL_URL=http://hop.home.lan:3170/graphql
VITE_BACKEND_WS_URL=wss://hop.home.lan:3170/graphql
VITE_BACKEND_API_URL=http://hop.home.lan:3170/v1
VITE_APP_TOS_LINK=http://example.com
VITE_APP_PRIVACY_POLICY_LINK=http://example.com
ENABLE_SUBPATH_BASED_ACCESS=false
```

### Environment

Production

### Version

Self-hosted

**Comment by StoneMoe** [comment]
In commit https://github.com/hoppscotch/hoppscotch/commit/afea75694fe0db497b0d2fc6cf1be05c636a38c8 which brought by https://github.com/hoppscotch/hoppscotch/pull/4233, we got a change about `* refactor: make containers run as non-root user in container`, which change the user from root to hoppuser.

https://github.com/hoppscotch/hoppscotch/blob/afea75694fe0db497b0d2fc6cf1be05c636a38c8/prod.Dockerfile#L41-L43

but the 80 port binding which introduced by https://github.com/hoppscotch/hoppscotch/pull/3449 apparently doesn't like that.
https://github.com/hoppscotch/hoppscotch/blob/1829c088ccb6cc1d8bcdadcca6e63b1d7ca2cc6e/packages/hoppscotch-backend/backend.Caddyfile#L1-L3

**Comment by AndrewBastin** [comment]
@StoneMoe do you use Podman (or any non-Docker) as your container runtime ? 

We were not able to reproduce this in both Docker and OrbStack.

**Comment by AndrewBastin** [comment]
UPDATE: Can repro this in Podman

Tested on NixOS 24.05 on x86_64

**Comment by StoneMoe** [comment]
> @StoneMoe do you use Podman (or any non-Docker) as your container runtime ?
> 
> We were not able to reproduce this in both Docker and OrbStack.

I'm using Docker, NOT non-Docker.
and using docker-compose via Portainer.
```
$ docker info

Client: Docker Engine - Community
 Version:    25.0.3
 Context:    default
 Debug Mode: false
 Plugins:
  buildx: Docker Buildx (Docker Inc.)
    Version:  v0.12.1
    Path:     /usr/libexec/docker/cli-plugins/docker-buildx
  compose: Docker Compose (Docker Inc.)
    Version:  v2.24.5
    Path:     /usr/libexec/docker/cli-plugins/docker-compose

Server:
 Containers: 19
  Running: 19
  Paused: 0
  Stopped: 0
 Images: 20
 Server Version: 25.0.3
 Storage Driver: overlay2
  Backing Filesystem: xfs
  Supports d_type: true
  Using metacopy: false
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: ae07eda36dd25f8a1b98dfbf587313b99c0190bb
 runc version: v1.1.12-0-g51d5e94
 init version: de40ad0
 Security Options:
  seccomp
   Profile: builtin
 Kernel Version: 3.10.0-1127.19.1.el7.x86_64
 Operating System: CentOS Linux 7 (Core)
 OSType: linux
 Architecture: x86_64
 CPUs: 64
 Total Memory: 62.44GiB
 Docker Root Dir: /data/docker
 Debug Mode: false
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

```

**Comment by StoneMoe** [comment]
I currently downgraded to 2024.7.0 as a workaround, and works fine.

**Comment by AndrewBastin** [comment]
@StoneMoe #4279 should probably fix this issue. If you are able to help verify whether the fix solves the issue, would appreciate it. The fix should be live in 2024.7.2 hopefully going live tomorrow.

**Comment by AndrewBastin** [comment]
@StoneMoe 2024.7.2 is live now, please do let me know if the issue is fixed in it.
